packages/react-native/src/Authenticator/Authenticator.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { useMemo } from 'react';
+
+import {
+  AuthenticatorProvider as Provider,
+  AuthenticatorComponentDefaults,
+  AuthenticatorMachineContext,
+  resolveAuthenticatorComponents,
+  useAuthenticator,
+  useAuthenticatorRoute,
+  useAuthenticatorInitMachine,
+  UseAuthenticator,
+} from '@aws-amplify/ui-react-core';
+
+import { AuthenticatorProps, FormHandlers } from './types';
+
+const DEFAULTS = {
+  // TODO: remove cast after adding missing components
+} as unknown as AuthenticatorComponentDefaults<FormHandlers>;
+
+const isAuthenticatedRoute = (route: UseAuthenticator['route']) =>
+  route === 'authenticated' || route === 'signOut';
```

Comment:
Are these the only two routes, or will we add more? If there's more, maybe we can put the routes in an array and use the `.includes` method